scripts/make-bundle

```diff
@@ -12,171 +12,196 @@ interface for those not familiar with the python
-
-
-def create_bootstrap_script(scratch_dir):
+
+
+PYTHON_VERSIONS = ["2.6", "2.7", "3.3", "3.4", "3.5", "3.6"]
+
+
+def pip(python_version):
+    if python_version == "2.6":
+        return "pip.__main__"
+    return "pip"
+
+
+def ensure_packages(requirements, *, output_dir, pythons=None):
+    if pythons is None:
+        pythons = PYTHON_VERSIONS
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # We want to loop over every version of Python we support and download
+        # or build wheels for them.
+        first = False
+        for python in sorted(pythons, reverse=True):
+            print("Downloading Dependencies for {}".format(python))
+
+            # Create our virtual environment.
+            venv_path = os.path.abspath(os.path.join(tmpdir, python))
+            python_path = os.path.join(venv_path, "bin", "python")
+            subprocess.run(
+                ["virtualenv", "-ppython{}".format(python), venv_path],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+
+            # Ensure that we have the latest versions of all of our tools
+            # installed in the virtual environment.
+            subprocess.run(
+                [python_path, "-m", pip(python), "install", "--upgrade",
+                 "pip"],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+            subprocess.run(
+                [python_path, "-m", pip(python), "install", "--upgrade",
+                 "setuptools", "wheel"],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+
+            if not first:
+                first = True
+                # We need to download and bundle virtualenv, but we need it to
+                # be the sdist since it won't actually be installed.
+                subprocess.run(
+                    [python_path, "-m", pip(python), "download",
+                     "--no-cache-dir", "--no-binary=virtualenv",
+                     "-d", output_dir, "virtualenv"],
+                    stdout=subprocess.PIPE,
+                    check=True,
+                )
+
+            # Use the virtual environment to build our wheels, doing this
+            # inside of a virtual environment ensures that we don't pollute the
+            # outer environment.
+            env = os.environ.copy()
+            env["CC"] = "/bin/false"
+            subprocess.run(
+                [python_path, "-m", pip(python), "wheel", "-w", output_dir,
+                 "--no-cache-dir"] + requirements,
+                stdout=subprocess.PIPE,
+                env=env,
+                check=True,
+            )
+
+            # We need to rename the PyYAML file, we've disabled the building of
+            # it's C extension so it is now pure Python, HOWEVER, because of
+            # the way it works wheel still thinks it is a binary file.
+            # TODO: It'd be great if this wasn't required.
```

Comment:
Yeah that's unfortunate. I am a little hesitant on relying on a regex search to determine the location of this file mainly because if the name of the file changes, it can cause issues. Is there any way we can better guarantee that we get a wheel file out of it?